File: javascript/matrix/matrix.js
export class Matrix {
  constructor(s) {
    this._rows = s.split('\n').map(row => row.split(' ').map(Number));
    this._columns = this._rows[0].map((_, i) => this._rows.map(row => row[i]));
  }

  get rows() {
    return this._rows;
  }

  get columns() {
    return this._columns;
  }
}

File: python/matrix/matrix.py
from typing import List


class Matrix:
    def __init__(self, matrix_string: str):
        rows = matrix_string.split('\n')
        self.matrix = [list(map(int, row.split(' '))) for row in rows]

    def row(self, index: int) -> List[int]:
        return self.matrix[index - 1]

    def column(self, index: int) -> List[int]:
        return [row[index - 1] for row in self.matrix]
